frontend/packages/kubevirt-plugin/src/k8s/objects/vm/vm/vm-clone.ts

```diff
@@ -0,0 +1,147 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { createBasicLookup, getName, getNamespace } from '@console/shared';
+import { VMKind } from '../../../../types/vm';
+import { getBasicID, joinIDs } from '../../../../utils';
+import {
+  getPvcAccessModes,
+  getPvcStorageClassName,
+  getPvcStorageSize,
+} from '../../../../selectors/pvc/selectors';
+import { DatavolumeTemplate } from '../datavolume-template';
+import { SafeVM } from './safe-vm';
+import {
+  getOperatingSystem,
+  getOperatingSystemName,
+  getVolumeDataVolumeName,
+  getVolumePersistentVolumeClaimName,
+} from '../../../../selectors/vm';
+import {
+  getDataVolumeAccessModes,
+  getDataVolumeStorageClassName,
+  getDataVolumeStorageSize,
+} from '../../../../selectors/dv/selectors';
+import { TEMPLATE_OS_NAME_ANNOTATION, TEMPLATE_VM_NAME_LABEL } from '../../../../constants/vm';
+
+export type CloneTo = {
```

Comment:
There was a discussion about that and this rule can be customized per plugin. 

The advantage of having the types (but mainly props) collocated with the class which uses them, is easier and faster code navigation - i.e. just looking at the relevant types

Reply:
> There was a discussion about that and this rule can be customized per plugin.
> 
> The advantage of having the types (but mainly props) collocated with the class which uses them, is easier and faster code navigation - i.e. just looking at the relevant types

This is exactly what @jtomasek prefers in Metal3 plugin code, I'm OK with it, if it makes you more productive.

When the file is small, ideally with just one component, putting types at the bottom isn't a big deal since the "distance" between type and its usage isn't that big. With bigger files containing lots of components, this might become an issue.